packages/frontend/src/components/UserSettings/UserAttributesPanel/UserAttributeModal.tsx

```diff
@@ -0,0 +1,201 @@
+        setInputError(undefined);
+        if (onClose) onClose();
+    };
+    const handleSubmit = async (data: CreateOrgAttribute) => {
+        // Input validation
+        if (!/^[a-z_][a-z0-9_]*$/.test(data.name)) {
+            setInputError(
+                `Invalid attribute name. Attribute name must contain only lowercase characters, '_' or numbers and it can't start with a number`,
+            );
+            return;
+        }
+        if (
+            allUserAttributes.some(
+                (attr) =>
+                    attr.name === data.name &&
+                    attr.uuid !== userAttribute?.uuid,
+            )
+        ) {
+            setInputError(`Attribute with the same name already exists`);
+            return;
+        }
+        if (userAttribute?.uuid) {
+            await updateUserAttribute(data);
+        } else {
+            await createUserAttribute(data);
+        }
+        handleClose();
+    };
+
+    const { data: orgUsers } = useOrganizationUsers();
+    return (
+        <Modal
+            opened={opened}
+            onClose={handleClose}
+            title={
+                <Title order={4}>
+                    {userAttribute ? 'Update' : 'Add'} user attribute
+                </Title>
+            }
+            size="lg"
+        >
+            <form
+                name="invite_user"
+                onSubmit={form.onSubmit((values: CreateOrgAttribute) =>
+                    handleSubmit(values),
+                )}
+            >
+                <Stack spacing="md">
+                    <TextInput
+                        name="name"
+                        label="Attribute name"
+                        placeholder="E.g. user_country"
+                        required
+                        {...form.getInputProps('name')}
+                    />
+                    <Text color="red" size="sm">
+                        {inputError}
+                    </Text>
+
+                    <Textarea
+                        name="description"
+                        label="Description"
+                        placeholder="E.g. The country where the user is querying data from."
+                        {...form.getInputProps('description')}
+                    />
+                    <Stack>
+                        <Text fw={500}>Assing to users</Text>
+
+                        {form.values.users?.length > 0 && (
+                            <Group>
+                                <Text w={200} fw={500}>
+                                    User email
+                                </Text>
+                                <Text fw={500}>Value</Text>
+                            </Group>
+                        )}
+
+                        {form.values.users?.map((user, index) => {
+                            return (
+                                <Group key={index}>
+                                    <Select
+                                        w={200}
```

Comment:
(note for self - check this in preview env)

Reply:
![CleanShot 2023-07-17 at 19 51 32@2x](https://github.com/lightdash/lightdash/assets/962095/cc518d75-69e2-405e-859a-2f2096a9193d)

Reply:
mm, using flexgrow does not make any difference `                                <Group key={index} sx={{flexGrow:1}}>`

![image](https://github.com/lightdash/lightdash/assets/1983672/7af14840-9360-4153-8d52-4db608b1d7a0)

Reply:
how do I fix the orange extra space ? should I wrap both groups in a stack ? or playing with margins

Reply:
you should apply `sx={{flexGrow:1}}` to individual input elements

Reply:
I'll check further on on that orange extra space

Reply:
Thanks ! updated 

![image](https://github.com/lightdash/lightdash/assets/1983672/e93a52af-7425-4bd2-96bd-c7e6ea66ac99)

And the orange extra space has been automaitcally fixed by using labels !